File: packages/camera/camera_avfoundation/example/ios/RunnerTests/CameraPropertiesTests.m
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

@import camera_avfoundation;
#if __has_include(<camera_avfoundation/camera_avfoundation-umbrella.h>)
@import camera_avfoundation.Test;
#endif
@import AVFoundation;
@import XCTest;

@interface CameraPropertiesTests : XCTestCase

@end

@implementation CameraPropertiesTests

#pragma mark - flash mode tests

- (void)testFCPGetAVCaptureFlashModeForPigeonFlashMode {
  XCTAssertEqual(AVCaptureFlashModeOff,
                 FCPGetAVCaptureFlashModeForPigeonFlashMode(FCPPlatformFlashModeOff));
  XCTAssertEqual(AVCaptureFlashModeAuto,
                 FCPGetAVCaptureFlashModeForPigeonFlashMode(FCPPlatformFlashModeAuto));
  XCTAssertEqual(AVCaptureFlashModeOn,
                 FCPGetAVCaptureFlashModeForPigeonFlashMode(FCPPlatformFlashModeAlways));
  XCTAssertThrows(FCPGetAVCaptureFlashModeForPigeonFlashMode(FCPPlatformFlashModeTorch));
}

#pragma mark - video format tests

- (void)testFCPGetPixelFormatForPigeonFormat {
  XCTAssertEqual(kCVPixelFormatType_32BGRA,
                 FCPGetPixelFormatForPigeonFormat(FCPPlatformImageFormatGroupBgra8888));
  XCTAssertEqual(kCVPixelFormatType_420YpCbCr8BiPlanarVideoRange,
                 FCPGetPixelFormatForPigeonFormat(FCPPlatformImageFormatGroupYuv420));
}

#pragma mark - device orientation tests

- (void)testFCPGetUIDeviceOrientationForPigeonDeviceOrientation {
  XCTAssertEqual(UIDeviceOrientationPortraitUpsideDown,
                 FCPGetUIDeviceOrientationForPigeonDeviceOrientation(
                     FCPPlatformDeviceOrientationPortraitDown));
  XCTAssertEqual(UIDeviceOrientationLandscapeLeft,
                 FCPGetUIDeviceOrientationForPigeonDeviceOrientation(
                     FCPPlatformDeviceOrientationLandscapeLeft));
  XCTAssertEqual(UIDeviceOrientationLandscapeRight,
                 FCPGetUIDeviceOrientationForPigeonDeviceOrientation(
                     FCPPlatformDeviceOrientationLandscapeRight));
  XCTAssertEqual(UIDeviceOrientationPortrait, FCPGetUIDeviceOrientationForPigeonDeviceOrientation(
                                                  FCPPlatformDeviceOrientationPortraitUp));
}

- (void)testFLTGetStringForUIDeviceOrientation {
  XCTAssertEqual(
      FCPPlatformDeviceOrientationPortraitDown,
      FCPGetPigeonDeviceOrientationForOrientation(UIDeviceOrientationPortraitUpsideDown));
  XCTAssertEqual(FCPPlatformDeviceOrientationLandscapeLeft,
                 FCPGetPigeonDeviceOrientationForOrientation(UIDeviceOrientationLandscapeLeft));
  XCTAssertEqual(FCPPlatformDeviceOrientationLandscapeRight,
                 FCPGetPigeonDeviceOrientationForOrientation(UIDeviceOrientationLandscapeRight));
  XCTAssertEqual(FCPPlatformDeviceOrientationPortraitUp,
                 FCPGetPigeonDeviceOrientationForOrientation(UIDeviceOrientationPortrait));
  XCTAssertEqual(FCPPlatformDeviceOrientationPortraitUp,
                 FCPGetPigeonDeviceOrientationForOrientation(-1));
}

@end

File: packages/camera/camera_avfoundation/example/ios/RunnerTests/CameraPropertiesTests.swift
// Copyright 2013 The Flutter Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import AVFoundation
import Foundation
import XCTest

@testable import camera_avfoundation

// Import Objective-C part of the implementation when SwiftPM is used.
#if canImport(camera_avfoundation_objc)
  import camera_avfoundation_objc
#endif

final class CameraPropertiesTests: XCTestCase {
  // MARK: - Flash Mode Tests

  func testGetAVCaptureFlashModeForPigeonFlashMode() {
    XCTAssertEqual(
      AVCaptureDevice.FlashMode.off,
      getAVCaptureFlashMode(for: .off))
    XCTAssertEqual(
      AVCaptureDevice.FlashMode.auto,
      getAVCaptureFlashMode(for: .auto))
    XCTAssertEqual(
      AVCaptureDevice.FlashMode.on,
      getAVCaptureFlashMode(for: .always))
  }

  // MARK: - Video Format Tests

  func testGetPixelFormatForPigeonFormat() {
    XCTAssertEqual(
      kCVPixelFormatType_32BGRA,
      getPixelFormat(for: .bgra8888))
    XCTAssertEqual(
      kCVPixelFormatType_420YpCbCr8BiPlanarVideoRange,
      getPixelFormat(for: .yuv420))
  }

  // MARK: - Device Orientation Tests

  func testGetUIDeviceOrientationForPigeonDeviceOrientation() {
    XCTAssertEqual(
      UIDeviceOrientation.portraitUpsideDown,
      getUIDeviceOrientation(for: .portraitDown)
    )
    XCTAssertEqual(
      UIDeviceOrientation.landscapeLeft,
      getUIDeviceOrientation(for: .landscapeLeft))
    XCTAssertEqual(
      UIDeviceOrientation.landscapeRight,
      getUIDeviceOrientation(for: .landscapeRight))
    XCTAssertEqual(
      UIDeviceOrientation.portrait,
      getUIDeviceOrientation(for: .portraitUp))
  }

  func testGetPigeonDeviceOrientationForUIDeviceOrientation() {
    XCTAssertEqual(
      FCPPlatformDeviceOrientation.portraitDown,
      getPigeonDeviceOrientation(for: .portraitUpsideDown))
    XCTAssertEqual(
      FCPPlatformDeviceOrientation.landscapeLeft,
      getPigeonDeviceOrientation(for: .landscapeLeft))
    XCTAssertEqual(
      FCPPlatformDeviceOrientation.landscapeRight,
      getPigeonDeviceOrientation(for: .landscapeRight))
    XCTAssertEqual(
      FCPPlatformDeviceOrientation.portraitUp,
      getPigeonDeviceOrientation(for: .portrait))
    // Test default case.
    XCTAssertEqual(
      FCPPlatformDeviceOrientation.portraitUp,
      getPigeonDeviceOrientation(for: .unknown))
  }
}
